bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java

```diff
@@ -355,12 +355,16 @@ private BArray retrieveArrayValues(TomlNode tomlValue, String variableName, Arra
     private BMap<BString, Object> retrieveRecordValues(TomlNode tomlNode, String variableName, Type type) {
         RecordType recordType;
         String recordName;
+        Module recordPkg;
         if (type.getTag() == TypeTags.RECORD_TYPE_TAG) {
             recordName = type.getName();
             recordType = (RecordType) type;
+            recordPkg = recordType.getPackage();;
         } else {
-            recordName = ((BIntersectionType) type).getConstituentTypes().get(0).getName();
+            Type originalType = ((BIntersectionType) type).getConstituentTypes().get(0);
```

Comment:
I think we can bring this logic to codegen side. May be we can do that in master branch